third_party/twig/twig/src/Node/IncludeNode.php

```diff
@@ -22,7 +22,7 @@
  */
 class IncludeNode extends Node implements NodeOutputInterface
 {
-    public function __construct(AbstractExpression $expr, AbstractExpression $variables = null, $only = false, $ignoreMissing = false, $lineno = null, $tag = null)
```

Comment:
Is this change by you or composer update?

Reply:
You can patch things in third_party/ folder, it will be overwritten by composer. Why do you need this?

Reply:
> Is this change by you or composer update?

The file "third_party/twig/twig/src/Node/IncludeNode.php" was updated in commit https://github.com/LimeSurvey/LimeSurvey/commit/5fa36197f5717588925a7efac24df7f54897ed08#diff-be0d7b384824e9eb33de24b7b4f85a48fd05a897acd1087df34052ac4e22d6a1.

Later that file was updated when I ran the composer install surely, returning to its original state.
Now, I reapplied the same commit.

It is good to ask ourselves: what is the correct way when you have to update something in the composer.json.
Because there are files (the autoload ones that are generated) that depend on the version of composer you use. And it is not clear to me if you have to use a specific one, or always use the last one. It sounds to me that varying the composer version can bring unexpected problems, right?

Reply:
Gah

Reply:
Yes, composer 2 might behave differently than composer 1. There's a dependency that breaks. See this thread: https://forum.yiiframework.com/t/problems-with-composer-2-and-bower-asset/130825/11